build.gradle

```diff
@@ -90,7 +93,28 @@ githubPages {
 }
 
 shadowJar {
-    classifier = 'all'
+    baseName = 'launchdarkly-client'
+    //no classifier means that the fat + shaded jar becomes the default artifact
+    classifier = ''
+
+    // Shade all jars except for launchdarkly
+    relocate('com', 'com.launchdarkly.shaded.com') {
```

Comment:
The ptential problem with shading Gson is that we return a Gson type (JsonElement) from our public interface: https://github.com/launchdarkly/java-client/blob/fa61ceb5f343c53db01f88b417a623575ab9fc58/src/main/java/com/launchdarkly/client/LDClient.java#L298-L298

This means that if a user's code already imports Gson this JsonElement is a different type than they might be expecting.